inc/compatibility/base.php

```diff
@@ -65,8 +72,16 @@ abstract class Base {
 	 */
 	public function __construct( $slug ) {
 		static::$plugin_slug = $slug;
-		static::$plugin_name =
-			get_plugin_data( WP_PLUGIN_DIR . '/' . static::$plugin_slug )['Name'] ?: static::$plugin_slug;
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+		$file = WP_PLUGIN_DIR . '/' . $slug;
+		if ( ! file_exists( $file ) ) {
+			static::$plugin_name = static::$plugin_slug;
```

Comment:
If I'm reading this correctly, this means that if a file named `${plugin_slug}/${slug}.php` doesn't exist, then the plugin name is just going to be the slug. ~~But a specific `$slug` can be passed when calling the base class, right? For example, I know Human Made has a pattern of naming plugins `${plugin_slug}/plugin.php`, which means it would break that conditional, but could I pass `plugin.php` into the base class when adding it to allow that pattern?~~ Edit: answered my own question. I'm not a huge fan of `directory-name/file-name.php` being the fallback, but that shouldn't be an issue if the plugins are added correctly, so I wouldn't change anything here at this time.

Sidenote: It would be nice if the `$slug` param was included in the PHPDoc.

Reply:
nice catch! added